test/hotspot/gtest/runtime/test_os.cpp

```diff
@@ -754,3 +754,62 @@ TEST_VM(os, dll_address_to_function_and_library_name) {
     }
   }
 }
+
+// Not a regex! Very primitive, just match:
+// "d" - digit
+// "a" - ascii
+// "." - everything
+// rest must match
+static bool very_simple_string_matcher(const char* pattern, const char* s) {
+  const size_t lp = strlen(pattern);
+  const size_t ls = strlen(s);
+  if (ls < lp) {
```

Comment:
Shouldn't this be `!=`?

Reply:
I wanted to be able to give prefix pattern which do not cover the whole string.